langstream-api-gateway/src/main/java/ai/langstream/apigateway/websocket/AuthenticationInterceptor.java

```diff
@@ -100,30 +133,64 @@ public AuthFailedException(String message) {
         }
     }
 
-    private Map<String, String> authenticate(GatewayRequestContext gatewayRequestContext)
+    private Map<String, String> authenticate(GatewayRequestContextImpl gatewayRequestContext)
             throws AuthFailedException {
+
         final Gateway.Authentication authentication =
                 gatewayRequestContext.gateway().authentication();
-        final Map<String, String> principalValues;
-        if (authentication != null) {
-            final String provider = authentication.provider();
-
-            final GatewayAuthenticationProvider authenticationProvider =
-                    GatewayAuthenticationProviderRegistry.loadProvider(
-                            provider, authentication.configuration());
-            final GatewayAuthenticationResult result =
-                    authenticationProvider.authenticate(gatewayRequestContext);
-            if (!result.authenticated()) {
-                throw new AuthFailedException(result.reason());
-            }
-            principalValues = result.principalValues();
+
+        if (authentication == null) {
+            return Map.of();
+        }
+
+        final GatewayAuthenticationResult result;
+        if (gatewayRequestContext.isAdminRequest()) {
+            if (!authentication.allowAdminRequests()) {
+                throw new AuthFailedException(
+                        "Gateway "
+                                + gatewayRequestContext.gateway().id()
+                                + " of tenant "
+                                + gatewayRequestContext.tenant()
+                                + " does not allow admin requests.");
+            }
+            String provider = gatewayRequestContext.adminCredentialsType();
+            if (provider == null && defaultProvider != null) {
+                provider = defaultProvider;
+            }
+            if (provider == null) {
+                throw new AuthFailedException("No admin auth provider specified");
+            }
+
+            final GatewayAdminAuthenticationProvider gatewayAdminAuthenticationProvider =
+                    adminAuthProviders.get(provider);
```

Comment:
I think that we shouldn't let the client decide which provider,
there will be only one configured, there is no need to have more